drivers/platform/stm32/stm32_gpio.c

```diff
@@ -277,7 +278,6 @@ int32_t stm32_gpio_direction_output(struct no_os_gpio_desc *desc,
 	HAL_GPIO_WritePin(extra->port, NO_OS_BIT(desc->number), (GPIO_PinState)value);
 
 	/* configure gpio with user configuration */
-	extra->gpio_config.Mode = GPIO_MODE_OUTPUT_PP;
```

Comment:
I think you left some things out, like this one:
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_gpio.c#L285

Reply:
The idea here is that _gpio_init has already set desc->extra to the initialization parameters (either given by application or default values) and here we just call init once again.